AsyncDisplayKit/Details/ASRangeControllerBeta.mm

```diff
@@ -225,6 +266,33 @@ - (void)_updateVisibleNodeIndexPaths
 #endif
 }
 
+#pragma mark - Notification observers
+
+- (void)registerForNotificationsIfNeeded
+{
+  if (!_didRegisterForNotifications) {
+    BOOL selfInterfaceState = [_dataSource interfaceStateForRangeController:self];
```

Comment:
Perhaps we should pass this in as an argument. Not a big deal, but it would eliminate several method calls.